api/v1beta1/openstackdataplanerole_types.go

```diff
@@ -82,3 +85,41 @@ func init() {
 func (instance OpenStackDataPlaneRole) IsReady() bool {
 	return instance.Status.Conditions.IsTrue(condition.ReadyCondition)
 }
+
+// Validate - validates the shared data between role and nodes
+func (instance OpenStackDataPlaneRole) Validate(nodes []OpenStackDataPlaneNode) error {
+	var errorMsgs []string
+
+	if len(instance.Spec.OpenStackAnsibleEERunnerImage) > 0 || len(instance.Spec.NetworkAttachments) > 0 {
+		for _, node := range nodes {
+			errorMsgs = AssertNodeAndRoleUniqueness(node, instance, errorMsgs)
+		}
+	}
+
+	// Compare nodes when role fields are empty
+	if len(instance.Spec.OpenStackAnsibleEERunnerImage) == 0 || len(instance.Spec.NetworkAttachments) == 0 {
+		nodeMap := make(map[string]OpenStackDataPlaneNode)
+
+		for _, node := range nodes {
+			if len(nodeMap["openStackAnsibleEERunnerImage"].Name) > 0 && len(node.Spec.OpenStackAnsibleEERunnerImage) > 0 && node.Spec.OpenStackAnsibleEERunnerImage != nodeMap["openStackAnsibleEERunnerImage"].Spec.OpenStackAnsibleEERunnerImage {
+				errorMsgs = append(errorMsgs, fmt.Sprintf("openStackAnsibleEERunnerImage mismatch between node: %s and node: %s", node.Name, nodeMap["openStackAnsibleEERunnerImage"].Name))
+			}
+
+			if len(nodeMap["networkAttachments"].Name) > 0 && len(node.Spec.NetworkAttachments) > 0 && !reflect.DeepEqual(node.Spec.NetworkAttachments, nodeMap["networkAttachments"].Spec.NetworkAttachments) {
+				errorMsgs = append(errorMsgs, fmt.Sprintf("networkAttachments mismatch between node: %s and node: %s", node.Name, nodeMap["networkAttachments"].Name))
+			}
+			// get the first non-empty
```

Comment:
I'm assuming the first time through the loop that the`nodeMap` is empty and so the first two conditions on lines 104 and 108 won't be true until after the first iteration. Thus, you're building the map in order to check that each item is the same as the previous item.